apluslms_roman/configuration.py

```diff
@@ -56,16 +64,54 @@ def find_from(cls, path, name=None, prefixes=None):
                     break
         else:
             if is_legacy_configuration(path):
-                container = cls.Container(config, allow_missing=True)
-                return cls(container, None, LEGACY_CONFIG, cls.version)
-            raise CourseConfigError("Couldn't find course configuration from {}. Expected to find one of these: {}".format(path, ', '.join(files)))
+                return cls.load(join(path, cls.DEFAULT_FILENAME), allow_missing=True)
+            raise ProjectConfigError((
+                _("Couldn't find course configuration from {}.\nExpected to find one of these: {}")
+            ).format(path, ', '.join(files)))
 
         return cls.load(config)
 
+    @classmethod
+    def load_from(cls, config):
+        if isfile(config):
+            return cls.load(config)
+        if '.' not in basename(config):
+            for prefix in cls.DEFAULT_PREFIXES:
+                filename = '%s.%s' % (config, prefix)
+                if isfile(filename):
+                    return cls.load(filename)
+        raise ProjectConfigError("Given file '{}' doesn't exist.".format(config))
+
+    def initialize_data(self):
+        super().initialize_data()
+        self.recursive_update(
+            LEGACY_CONFIG if is_legacy_configuration(self.dir)
+            else DEFAULT_CONFIG
+        )
+
+    def add_step(self, step):
+        if 'name' in step:
+            if step['name'].lower() in self.steps_by_name:
+                raise ValueError("A step with the name '%s' already exists." % step['name'])
+        self.steps.append(step)
```

Comment:
You probably need to ensure `steps` is in the self container. In case where `steps` is not in the container, then we would effectively do `[].append(step)`. Of course, this is mostly handled by schema (for now), so this shouldn't ever happen. Though, schema probably shouldn't require at least one item in steps list.

```
if 'steps' not in self:
  self['steps'] = [step]
else:
  self['steps'].append(step)
```

It's possible that `self.setdefault('steps', [])` in `def steps(self):` would be a better solution and it might even work correctly. Would set 'steps' to the ChangesDict default set and when an value is added to the list, it should move the list from defaults to data (at least ChangesList should work that way if it doesn't currently).

Reply:
should I change schema to not require steps? although what is the point of a course config if it doesn't have steps in it?

Reply:
Nothing, except it would tell roman that this is a course/project folder, so it doesn't raise error (no config found). The legacy config tries to do sphinx build and there are courses that use only yaml files.

Later on, there might be service definitions, which might be the only thing required in that case. But yeah, currently it would seem silly to have config without any steps.

That said, I guess the `steps` key shouldn't be required either.